File: e2e/playwright-typescript/tests/task-table/task-table.axe.spec.ts
import { test } from '@/fixtures';
import { TaskContext } from '@/context/TaskContext';
import * as allure from "allure-js-commons";
import { AllureEpic } from '@/data/AllureEpic';

test.beforeAll(async () => {
  await allure.epic(AllureEpic.ACCESSIBILITY);
  await allure.feature('Task table');
  await allure.tms('98');
});

test.describe('Task table - accessibility', () => {
  test.beforeEach(async ({ support }) => {
    await support.mock.api.getTasks([
      new TaskContext().createTaskResponse(),
      new TaskContext().createTaskResponse(),
    ]);
  });

  test('should have no accessibility violations on task table', { tag: '@accessibility' }, async ({ step, validate }) => {
    // given
    await step.navigation.openMainPage();

    // when
    const results = await step.accessibility.analyze();

    // then
    await validate.accessibility.hasNoViolations(results);
  });
});

File: e2e/capybara-ruby/spec/task_table/task_table_axe_spec.rb
# frozen_string_literal: true

require 'spec_helper'

require_relative '../../context/task_context'
require_relative '../../data/allure_epic'

RSpec.describe 'Task table - accessibility', type: :system, accessibility: true,
                                              epic: AllureEpic::ACCESSIBILITY,
                                              feature: 'Task table',
                                              tms: '98' do
  before do
    support.mock.api.get_tasks([
                                 TaskContext.new.create_task_response,
                                 TaskContext.new.create_task_response
                               ])
  end

  it 'should have no accessibility violations on task table' do
    # given
    step.navigation.open_main_page

    # when
    results = step.accessibility.analyze

    # then
    validate.accessibility.has_no_violations(results)
  end
end
